packages/hardhat-chai-matchers/test/bigNumber.ts

```diff
@@ -103,7 +103,7 @@ describe("BigNumber matchers", function () {
     });
 
     interface FailureCase extends SuccessCase {
-      msg: string;
+      msg: string | RegExp;
```

Comment:
Note that this line of code is just the way to faciltiate the changes like this: https://github.com/NomicFoundation/hardhat/pull/2746/files#diff-a245c4b4d965ae1e3c68c6146a5402f35d3e0851a17a236094c667ecf56d3faeR118

Honestly I'm not sure what caused this change, but I'm going to guess it was Chai, because the symptom is that standard types like `Map` are being displayed differently in error messages, as you can see here: https://github.com/NomicFoundation/hardhat/runs/6528952333?check_suite_focus=true#step:6:19220

Reply:
Makes sense, thanks for the explanation